app/src/main/java/protect/card_locker/LetterBitmap.kt

```diff
@@ -43,98 +53,90 @@ class LetterBitmap {
      * @param backgroundColor    (optional) color to use for background.
      * @param textColor          (optional) color to use for text.
      */
-    public LetterBitmap(Context context, String displayName, String key, int tileLetterFontSize,
-                        int width, int height, Integer backgroundColor, Integer textColor) {
-        TextPaint paint = new TextPaint();
+    init {
+        val paint = TextPaint()
 
         if (textColor != null) {
-            paint.setColor(textColor);
+            paint.color = textColor
         } else {
-            paint.setColor(Color.WHITE);
+            paint.color = Color.WHITE
         }
 
-        paint.setTextAlign(Paint.Align.CENTER);
-        paint.setAntiAlias(true);
-        paint.setTextSize(tileLetterFontSize);
-        paint.setTypeface(Typeface.defaultFromStyle(Typeface.BOLD));
+        paint.textAlign = Paint.Align.CENTER
+        paint.isAntiAlias = true
+        paint.textSize = tileLetterFontSize.toFloat()
+        paint.typeface = Typeface.defaultFromStyle(Typeface.BOLD)
 
-        if (backgroundColor == null) {
-            mColor = getDefaultColor(context, key);
-        } else {
-            mColor = backgroundColor;
-        }
+        mColor = backgroundColor ?: getDefaultColor(context, key)
 
-        mBitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
-        String firstChar = displayName.substring(0, 1).toUpperCase();
-        int firstCharEnd = 2;
-        while (firstCharEnd <= displayName.length()) {
+        this.letterTile = createBitmap(width, height)
```

Comment:
This is a completely different Bitmap function, please don't change the meaning of the code, it just needs to be turned from Java to Kotlin, it shouldn't change behaviour.